### PR TITLE
Clean up more web platform tests

### PR DESCRIPTION
**Pull request by domenic**
Use promise_test instead of async_test; lint them; use ES2015 features

I don't expect anyone to review this as it's largely mechanical changes but I'll leave it up for a couple days just in case.

### CODE REVIEW

File: reference-implementation/web-platform-tests/readable-streams/brand-checks.js [hunk 1]
```diff
@@ -5,118 +5,147 @@ if (self.importScripts) {
...
+    methodRejects(t, ReadableStreamReader.prototype, 'read', fakeReadableStreamReader()),
+    methodRejects(t, ReadableStreamReader.prototype, 'read', realReadableStream(), true)
+  ]);
+
+}, 'ReadableStreamReader.prototype.read enforces a brand check');
+
+test(() => {
+
+  methodThrows(ReadableStreamReader.prototype, 'releaseLock', fakeReadableStreamReader());
+  methodThrows(ReadableStreamReader.prototype, 'releaseLock', realReadableStream(), true);
```
Comment (tyoshino) [inline reference-implementation/web-platform-tests/readable-streams/brand-checks.js]: methodThrows doesn't take endTest parameter?

File: reference-implementation/web-platform-tests/readable-streams/general.js [hunk 2]
```diff
@@ -1,758 +1,831 @@
...
-        assert_equals(timesCalled, 4, 'pull() should have been called four times');
-    });
+      size() {
+        return 1;
+      },
+      highWaterMark: 4
+    }
+  );
+
+  return ready.then(() => {
```
Comment (tyoshino) [inline reference-implementation/web-platform-tests/readable-streams/general.js]: delay(10)? it's unnecessary?

File: reference-implementation/web-platform-tests/readable-streams/readable-stream-reader.js [hunk 3]
```diff
@@ -5,520 +5,480 @@ if (self.importScripts) {
...
+  const rs = new ReadableStream({
+    start(c) {
+      controller = c;
+    }
+  });
+
+  const reader1 = rs.getReader();
+
+  promiseAsserts.push(
+    promise_rejects(t, theError, reader1.closed)
```
Comment (tyoshino) [inline reference-implementation/web-platform-tests/readable-streams/readable-stream-reader.js]: push reader1.read()?

File: reference-implementation/web-platform-tests/readable-streams/general.js (hunk 2 above)
Comment (domenic) [inline reference-implementation/web-platform-tests/readable-streams/general.js]: Yeah, it is.

File: reference-implementation/web-platform-tests/readable-streams/readable-stream-reader.js (hunk 3 above)
Comment (domenic) [inline reference-implementation/web-platform-tests/readable-streams/readable-stream-reader.js]: Thanks, good find

File: reference-implementation/web-platform-tests/readable-streams/brand-checks.js (hunk 1 above)
Comment (domenic) [inline reference-implementation/web-platform-tests/readable-streams/brand-checks.js]: Thanks, fixed here and elsewhere in the file

**Pull request merged**